processing/src/main/java/io/druid/query/aggregation/post/ExpressionPostAggregator.java

```diff
@@ -70,22 +75,33 @@ public ExpressionPostAggregator(
       @JacksonInject ExprMacroTable macroTable
   )
   {
+    this(name, expression, ordering, macroTable, null);
+  }
+
+  /**
+   * Constructor for {@link #decorate(Map)}.
+   */
+  private ExpressionPostAggregator(
+      final String name,
+      final String expression,
+      final String ordering,
+      final ExprMacroTable macroTable,
+      final Map<String, Function<Object, Object>> finalizers
+  )
+  {
     Preconditions.checkArgument(expression != null, "expression cannot be null");
 
     this.name = name;
     this.expression = expression;
     this.ordering = ordering;
     this.comparator = ordering == null ? DEFAULT_COMPARATOR : Ordering.valueOf(ordering);
     this.macroTable = macroTable;
+    this.finalizers = finalizers != null ? ImmutableMap.copyOf(finalizers) : ImmutableMap.of();
```

Comment:
Since this is an internal constructor, suggested to elimitate nullability (and rather check for non-null), and pass `ImmutableMap.of()` in another constructor.